daemon/pod/decommission.go

```diff
@@ -421,96 +402,97 @@ func (p *XPod) stopAllContainers(graceful int) error {
 	}
 
 	var (
-		cMap  = make(map[string]bool, len(p.containers))
 		cList = make([]string, 0, len(p.containers))
 	)
 
 	for cid := range p.containers {
-		cMap[cid] = true
 		cList = append(cList, cid)
 	}
 
-	err, eMap := p.stopContainers(cList, cMap, graceful)
+	err := p.stopContainers(cList, graceful)
 	if err != nil {
 		p.Log(ERROR, "exception during stop all containers: %v", err)
 	}
-	if len(eMap) > 0 {
-		e := fmt.Errorf("some container failed during stopping %#v", eMap)
-		p.Log(ERROR, eMap)
-		if err == nil {
-			err = e
-		}
-	}
 
 	return err
 }
 
-func (p *XPod) stopContainers(cList []string, cMap map[string]bool, graceful int) (error, map[string]error) {
-	type containerException struct {
-		id  string
-		err error
-	}
-
+func (p *XPod) stopContainers(cList []string, graceful int) error {
 	p.Log(INFO, "begin stop containers %s", cList)
-	resChan := p.sandbox.WaitProcess(true, cList, graceful)
-	errChan := make(chan *containerException, 1)
 
+	future := utils.NewFutureSet()
+	waitTime := time.Duration(graceful) * time.Second
+	if graceful == 0 {
```

Comment:
If `graceful == 0`, it just wait 5s and won't kill again(return timeout error); if kill with image specified signal(graceful != 0), it will wait graceful time to kill with SIGKILL again.
Is this 5 seconds too long?